### PR TITLE
[OTS-484] Add selected and hover states to toolbar items

### PR DESCRIPTION
**Pull request by adrice727**
👍

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/opentok/annotation-acc-pack/pull/94" target="_blank">ReviewNinja

File: js/opentok.js-annotations/dist/opentok-annotation.js [hunk 1]
```diff
@@ -160,6 +160,30 @@
...
+        });
+
+        // If the selected item is a shape, update the shapes subpanel item
+        var el = document.getElementById(id);
+        if (el.parentElement.classList.contains('shapes')) {
+          document.getElementById('OT_shapes').classList.add('selected');
+        } else {
+          el.classList.add('selected');
+        }
+      }
```
Comment (Lucashuang0802) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: 👍

File: js/opentok.js-annotations/dist/opentok-annotation.js [hunk 2]
```diff
@@ -195,11 +220,6 @@
 
     this.undo = function () {
       undoLast(false, self.session.connection.connectionId);
-      if (self.session) {
-        self.session.signal({
-          type: 'otAnnotation_undo'
-        });
-      }
```
Comment (Lucashuang0802) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: We don't signal undo? I'm afraid we need it from mobile.

File: js/opentok.js-annotations/dist/opentok-annotation.js [hunk 3]
```diff
@@ -160,6 +160,30 @@
...
+        var id = item.id;
+
+        // Undo and clean are actions, not items that can be selected
+        if (id === 'OT_undo' || id === 'OT_clear') {
+          return;
+        }
+        // Remove the 'selected' class from all buttons and panels
+        Array.from(document.getElementsByClassName('annotation-btn')).forEach(function (btn) {
+          btn.classList.remove('selected');
+        });
```
Comment (Lucashuang0802) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: Do you really need to traverse the whole array? There is only one item that has `selected` true, right?

File: js/opentok.js-annotations/dist/opentok-annotation.js [hunk 4]
```diff
@@ -1580,6 +1601,26 @@
...
+         *       states.
+         */
+        var toggleColorsHover = function (hover) {
+          var action = hover ? 'add' : 'remove';
+          document.getElementById('OT_toolbar').classList[action]('colors-hover');
+        };
+        var colors = document.getElementById('OT_colors');
+        colors.addEventListener('mouseenter', function () { toggleColorsHover(true); });
+        colors.addEventListener('mouseleave', function () { toggleColorsHover(false); });
+        /** End color picker hover state */
```
Comment (Lucashuang0802) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: 👍

File: js/opentok.js-annotations/dist/opentok-annotation.js (hunk 2 above)
Comment (adrice727) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: Just web -> web right now.

**Comment by Lucashuang0802** [comment]
👍

**Comment by adrice727** [comment]
👍

File: js/opentok.js-annotations/dist/opentok-annotation.js [hunk 5]
```diff
@@ -160,13 +160,37 @@
...
+      /**
+       * Update classes for toolbar items
+       */
+      var updateSelected = function () {
+
+        var current = document.getElementById(self.selectedItem.id);
+        var shapesBtn = document.getElementById('OT_shapes');
+
+        // Remove the 'selected' class from the currently selected item (or parent)
+        var current = document.getElementById(self.selectedItem.id);
```
Comment (Lucashuang0802) [inline js/opentok.js-annotations/dist/opentok-annotation.js]: This line is duplicate with LOC168?

**Comment by Lucashuang0802** [comment]
LGTM

**Comment by Lucashuang0802** [comment]
LGTM

**Comment by adrice727** [comment]
LGTM

**Pull request merged**